core/src/main/java/org/elasticsearch/common/network/NetworkAddress.java

```diff
@@ -70,6 +73,22 @@ public static String format(InetAddress address) {
     }
 
     /**
+     * Returns a string representation of an {@link InetAddress} that is
+     * compatible with sorting and {@link InetAddress#getByName(String)}.
+     */
+    public static String formatToSortableString(InetAddress ip) {
```

Comment:
Is this really the best place for this? Seems it has nothing to do with actual networking, but just with the ip mapping, so it should probably go there? And does it really need to be public or can it be an impl detail of the mapper?